torch/csrc/jit/codegen/cuda/dispatch.h

```diff
@@ -75,49 +75,10 @@ class TernaryOp;
 class ReductionOp;
 class BroadcastOp;
 
-// Kernel IR
-namespace kir {
-
-class Bool;
-class Float;
-class Half;
-class Int;
-class NamedScalar;
-
-class IterDomain;
-class TensorDomain;
-class TensorView;
-
-class UnaryOp;
-class BinaryOp;
-class TernaryOp;
-class ReductionOp;
-class BroadcastOp;
-
-class TensorIndex;
-class Allocate;
-class ForLoop;
-class IfThenElse;
-class GridReduction;
-class Sync;
-
-} // namespace kir
-
-/*
- * By default, all IR nodes are handled in this dispatch, and will call an empty
- * function on all nodes.
- */
-class TORCH_CUDA_API OptOutConstDispatch {
+// By default, all IR nodes are handled in this dispatch, and will call an empty
+// function on all nodes.
+class TORCH_CUDA_API OptOutConstDispatch : public PolymorphicBase {
```

Comment:
Why inheriting PolymorphicBase? Is every class supposed to inherit this class? Is it necessary for this PR?

Reply:
Since I had to make deep changes to dispatch.h I took the opportunity to clean a few obvious things. Inheriting from `PolymorphicBase` eliminates the need from defining a virtual destructor in every dispatch variant.